resources/main.mjs

```diff
@@ -19,9 +20,11 @@ class MainBenchmarkClient {
     _metrics = Object.create(null);
     _steppingPromise = null;
     _steppingResolver = null;
+    _dataProvider = null;
 
-    constructor() {
-        window.addEventListener("DOMContentLoaded", () => this.prepareUI());
```

Comment:
we're not waiting for DOMContentLoaded anymore?

Or is it because of the `await` above we're _usually_ past this event... but this is racy so we should probably keep it but _also_ check for `document.readyState`...